### ISSUE TITLE
Change `${matched}` to match over results of `git ls-files`

### ISSUE THREAD

**Issue by atifaziz** [alirezanet/Husky.Net#17]
`${matched}` casts too wide a net and scans/matches too many files by default. It's not usually what one wants because it can potentially end up returning hidden files and scanning the object database under `.git/objects` that could be very large. I would propose the default behaviour to return matches over files under version control, i.e. those returned by `git ls-files`. This way, 
one shouldn't have to [_be (too) careful with the variable_](https://github.com/alirezanet/Husky.Net/blame/0697290bc00b91d5862f05b88e9ca987e132cf4d/README.md#L205).

**Comment by alirezanet** [comment]
Hi, yes I agree `git ls-files` makes more sense in 90 percent of scenarios but what if people want to do some cleanup using task runner (that requires ignored files), I couldn't find a solution for this that yet.  maybe I should use `git ls-files` by default when include and exclude are empty and use ``Directory.GetFiles` when user entered some globs... what do you think?

**Comment by alirezanet** [comment]
Or another solution could be adding a separate variable like `${git-matched}`

**Comment by atifaziz** [comment]
> but what if people want to do some cleanup using task runner (that requires ignored files), I couldn't find a solution for that yet.

It feels like Husky.Net is trying to be two things, Git hooks made easy _and_ a general-purpose task runner? If the task runner is purely for facilitating the hooks then I don't expect people to be dealing with ignored files or running clean-up tasks.

> maybe I should use `git ls-files` by default when include and exclude are empty and use `Directory.GetFiles` when the task had some globs...

That could be even more confusing. The behviour should could surprise people to see different results with and without the globs.

**Comment by alirezanet** [comment]
> It feels like Husky.Net is trying to be two things,

Yes, I'm planning to support more

>  I don't expect people to be dealing with ignored files or running clean-up tasks.

I personally,  use something like this in one of the projects, e.g. using the pre-push hook as an event to do remove the obj/bin folder that is excluded.

> It feels like Husky.Net is trying to be two things,

Except ignoring the `.git` folder that is necessary what is your suggestion?

**Comment by alirezanet** [comment]
I've added a `${committed}` variable, probably this is what you looking for.

**Comment by atifaziz** [comment]
`${committed}` could be misleading since `git ls-files` will include staged files. Perhaps something as simple as `${git-ls-files}` so that there is not guess work needed? In fact, I would say that `${matched}` should be renamed to `${all}`, `${files}` or `${*}` if you wish to keep it bound to the file system rather than Git.

Ultimately, if you really want to be open-ended then another idea would be to let the user define the source of the file list as another command. You could introduce the notion of _hidden commands_ in `task-runner.json`. That is, any command whose name starts with `.` won't be run by default _but_ could be used as a variable and its output becomes the file list. Here is how one could image a task runner JSON with two hidden commands (`.git-ls-files` and `.files`) and two test commands (`test-git-ls-files` and `test-files`) could look like:

```json
{
   "tasks": [
      {
         "name": ".git-ls-files",
         "command": "git",
         "args": [ "ls-files" ]
      },
      {
         "name": ".files",
         "command": "find",
         "args": [ ".", "-type", "f", "-name", "*" ],
         "windows": {
            "command": "cmd",
            "args": [ "/c", "dir", "/s", "/b" ]
         }
      },
      {
         "name": "test-ls-files",
         "command": "printf",
         "args": [ "%s\\n", "${.git-ls-files}" ],
         "include": [ "**/*.cs", "**/*.vb" ],
         "windows": {
            "command": "cmd",
            "args": [ "/c", "echo", "${.git-ls-files}" ]
         }
      },
      {
         "name": "test-files",
         "command": "printf",
         "args": [ "%s\\n", "${.git-ls-files}" ],
         "include": [ "**/*.cs", "**/*.vb" ],
         "windows": {
            "command": "cmd",
            "args": [ "/c", "echo", "${.files}" ]
         }
      }
   ]
}
```

So you could provide some built-in ones (like `${staged}` and `${lastCommit}` that are already there) and then leave the stage open for other, more custom and advanced scenarios.

**Comment by alirezanet** [comment]
Considering your suggestions I changed the variable section in general. check it out [here](https://github.com/alirezanet/Husky.Net#arg-variables).

This issue should be fixed via #18 PR. in `v0.2.0`.

thanks for the feedback.
